src/Pact/Types/RowData.hs

```diff
@@ -128,11 +129,29 @@ instance ToJSON RowData where
   toJSON (RowData v m) = object
       [ "$v" .= v, "$d" .= m ]
 
+newtype OldPactValue =
+  OldPactValue { unP :: PactValue }
+
+instance FromJSON OldPactValue where
```

Comment:
Does this really work? Seems like it will only affect toplevel while nested will use `PactValue` decoder, since you haven't redefined the containers to contain `OldPactValue`

Reply:
Made it into its own datatype :+1: